apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/TxnHistoryTable.tsx

```diff
@@ -1,153 +1,199 @@
+
+      const { data: res } = await fetchGenericSubgraph<FetchTxnsResponse>(
+        env.subgraph.templeV2,
+        `{
+            strategies${strategyQuery} {
+              transactionCount
+            }
+            _meta {
+              block {
+                number
+              }
+            }
+          }`
+      );
+      if (!res) return;
+      let txCountTotal = 0;
+      res.strategies.map(s=> txCountTotal += s.transactionCount);
+      setTotalPages(Math.ceil(txCountTotal / rowsPerPage));
+      setBlockNumber(res._meta.block.number);
+    })();
+  }, [dashboardType, rowsPerPage, filter, restartTable]);
+
+  const fetchTransactions = useCallback(
+    async (strategyType: StrategyType): Promise<Transactions> => {
+      
+      const strategyQuery = strategyType === 'All' ? `` : `where: { name: "${strategyType}" }`;
+      const blockNumberQueryParam = blockNumber > 0 ? `block: { number: ${blockNumber} }` : ``;
+      let strategyAndBlockQuery = '';
+      if (blockNumberQueryParam.length > 0 || strategyQuery.length> 0){
+        strategyAndBlockQuery = `(${blockNumberQueryParam} ${strategyQuery})`
+      }
+      
+      const paginationQuery = `skip: ${(currentPage - 1) * rowsPerPage} first: ${rowsPerPage}`;
+      
+      const dateNowSecs = Math.round( Date.now() / 1000);
+      const filterQuery = `where: { timestamp_gt: ${ dateNowSecs - txHistoryFilterTypeToSeconds(filter)} }`;
+
+      const { data: res } = await fetchGenericSubgraph<FetchTxnsResponse>(
+        env.subgraph.templeV2,
+        `{
+          strategies${strategyAndBlockQuery} {
+            name
+            id
+            transactions(orderBy: timestamp, orderDirection: desc ${paginationQuery} ${filterQuery}) {
+              hash
+              amount
+              amountUSD
+              id
+              from
+              kind
+              timestamp
+            }
+          }
+        }`
+      );
+      if (!res) return [];
+      const txns: Transactions = [];
+      for (const strategy of res.strategies) {
+        strategy.transactions.map((t) => txns.push(t));
+      }
+      return txns;
+    },
+    [blockNumber, currentPage, rowsPerPage, filter]
+  );
 
-  const lastRow = currentPage * rowsPerPage;
-  const firstRow = lastRow - rowsPerPage;
-  const totalPages = Math.ceil(data.length / rowsPerPage);
-  const currentData = data.slice(firstRow, lastRow);
-
-  // Fetch data
+  // Fetch strategies tx data
   useEffect(() => {
-    const fetchData = async () => {
+    (async () => {
+      setIsLoading(true);
       try {
-        // Replace with your API endpoint
-        // const response = await fetch('https://api.yourdomain.com/data-endpoint');
-        // const result = await response.json();
-
-        // setData(result.data);
-        // TODO: USE filter
-        console.log('filter', filter);
+        const txns = await fetchTransactions(dashboardTypeToStrategyType(dashboardType));
```

Comment:
I'm not a big fan of doing frontend queries with `useEffect/useCallback`, but following current src. Would prefer `react-query` instead :)